### PR TITLE
feat: support configuring linaria with a config file

### PR DESCRIPTION
**Pull request by satya164**
The babel preset is used in several ways:
- babel config
- webpack loader
- rollup plugin
- stylelint preprocessor

Having a config file makes it possible to configure it in one place instead of configuring separately for each tool.

### CODE REVIEW

File: src/rollup.js [hunk 1]
```diff
@@ -6,10 +6,11 @@ const { createFilter } = require('rollup-pluginutils');
 const transform = require('./transform');
 const slugify = require('./slugify');
 
-type RollupPluginOptions = PluginOptions & {
+type RollupPluginOptions = {
   include?: string | string[],
   exclude?: string | string[],
   sourceMap?: boolean,
+  ...$Shape<PluginOptions>,
```
Comment (thymikee) [inline src/rollup.js]: What do you think about making a separate entry for bundlers in the config, so they don't overlap with each other (like with exclude vs ignore)?
We could have global options available for all bundlers, and specific ones if needed.

**Comment by zamotany** [comment]
LGTM, but does that mean the user can add their own babel presets/plugins so that linaria will use them? For example, does this allow to support `babel-plugin-module-resolver`?

File: src/rollup.js (hunk 1 above)
Comment (satya164) [reply src/rollup.js]: @thymikee we could. let's talk about it. probably out of scope for this PR.

**Comment by satya164** [comment]
@zamotany I only support the same options as previous in the config file. though regarding babel transforms, I believe they should be respected during module evaluation if they are in the babel config since we don't disable `babelrc`: https://github.com/callstack/linaria/blob/master/src/babel/evaluate.js#L159

File: src/rollup.js (hunk 1 above)
Comment (thymikee) [reply src/rollup.js]: Yup, not strictly related, just wanted to spark the discussion. e.g.

```js
module.exports = {
  displayName: true,
  bundler: {
    rollup: {
  	  sourceMap: true
    },
    webpack: { 
      // ... 
    }
  }

}
```

Comment (satya164) [reply src/rollup.js]: should the configuration file support configuring bundlers? 🤔

Comment (thymikee) [reply src/rollup.js]: Maybe not, but it's weird that Rollup config is merged with linaria's :|

Comment (satya164) [reply src/rollup.js]: Yeah. We already did that for webpack so it's the same here. Maybe we should remove passing linaria's options from bundler config or make it a separate key there?

Comment (thymikee) [reply src/rollup.js]: Mhm, we should only pass the options that make sense to this very bundler.

**Pull request merged**